src/filterScreenings.js

```diff
@@ -0,0 +1,61 @@
+/*Filters out ten screenings in the next five days or removes the last day's screenings incase
+the number of screenings pushes the total number of screenings to more than ten*/
+export function filterScreenings(data) {
+  let filteredScreenings = [];
+  data.forEach((screening) => {
+    
+    // Setting a starting data for the next day to start displaying screenings
+    const currentScreeningDate = new Date(screening.attributes.start_time);
+    const startingPoint = new Date(Date.now() + 86400000);
+    startingPoint.setHours(0,0,0,0);
+
+    //Setting an end date for the screenings to be displayed
+    const endPoint = new Date();
+    endPoint.setDate(startingPoint.getDate()+7);
```

Comment:
I'm pretty sure this will break at some points in a month, in some timezones.

Let's think about the last day of a month, e.g. March 31st, at 13:37 in London (the `0` timezone`).

1. After line 10, `startingPoint` will be `2023-04-01T00:00:00`.
2. On line 13, `endPoint` becomes `2023-03-31T13:37:00`
3. On line 14, the date of `endPoint` is set to `2023-03-08`, which is not what you're expecting

Why does this happen?

Because the date of `endPoint` will be set to the value of `startingPoint.getDate()` plus 7. The value of `startingPoint.getDate()` will be 1 (first of April, as assigned in step 1 above). Adding 7 gives you 8, so the date is changed to 8, but the month doesn't change.

It should be easy to fix this by using `endPoint.getDate()` instead of `startingPoint.getDate()` on line 14. That way the date will always be 7 days after the current date.

Conclusion: Dates are difficult! That's why we always try to test them extensively.